src/transformers/models/beit3/convert_beit3_unilm_to_pytorch.py

```diff
@@ -0,0 +1,389 @@
+        label2id=label2id,
+    )
+
+
+def get_large_config_visual_reasoning(img_size):
+    id2label = {0: "False", 1: "True"}
+    label2id = {v: k for k, v in id2label.items()}
+    return Beit3Config(
+        hidden_size=1024,
+        num_hidden_layers=24,
+        num_attention_heads=16,
+        intermediate_size=1024 * 4,
+        num_labels=2,
+        image_size=img_size,
+        normalize_before=True,
+        encoder_normalize_before=True,
+        id2label=id2label,
+        label2id=label2id,
+    )
+
+
+def get_base_config_captioning(img_size):
+    return Beit3Config(
+        intermediate_size=768 * 4,
+        num_labels=2,
+        image_size=img_size,
+        normalize_before=True,
+        encoder_normalize_before=True,
+    )
+
+
+def get_large_config_captioning(img_size):
+    return Beit3Config(
+        hidden_size=1024,
+        num_hidden_layers=24,
+        num_attention_heads=16,
+        intermediate_size=1024 * 4,
+        num_labels=2,
+        image_size=img_size,
+        normalize_before=True,
+        encoder_normalize_before=True,
+    )
+
+
+def get_base_config_image_text_retrieval(img_size):
+    return Beit3Config(
+        intermediate_size=768 * 4,
+        num_labels=2,
+        image_size=img_size,
+        normalize_before=True,
+        encoder_normalize_before=True,
+    )
+
+
+def get_large_config_image_text_retrieval(img_size):
+    return Beit3Config(
+        hidden_size=1024,
+        num_hidden_layers=24,
+        num_attention_heads=16,
+        intermediate_size=1024 * 4,
+        num_labels=2,
+        image_size=img_size,
+        normalize_before=True,
+        encoder_normalize_before=True,
+    )
+
+
+def rename_keys(model_state_dict, beit3_model_type):
+    new_state_dict = copy.deepcopy(model_state_dict)
+    rename_key_names = rename_key_mappings.keys()
+    for key in model_state_dict:
+        current_key = copy.deepcopy(key)
+        for rename_key in rename_key_names:
+            if rename_key in current_key:
+                val = new_state_dict.pop(current_key)
+                new_state_dict[current_key.replace(rename_key, rename_key_mappings[rename_key])] = val
+                current_key = current_key.replace(rename_key, rename_key_mappings[rename_key])
+
+    if beit3_model_type == "vqa":
+        keys_to_be_renamed_for_vqa = [
+            "pooler.norm.weight",
+            "pooler.norm.bias",
+            "pooler.dense.weight",
+            "pooler.dense.bias",
+        ]
+        for key in keys_to_be_renamed_for_vqa:
+            new_state_dict[f"beit3.{key}"] = new_state_dict[key]
+            del new_state_dict[key]
+    return new_state_dict
+
+
+def prepare_img():
+    url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+    im = Image.open(requests.get(url, stream=True).raw)
+    return im
+
+
+def get_tokenizer():
+    return XLMRobertaTokenizer.from_pretrained("https://github.com/addf400/files/releases/download/beit3/beit3.spm")
```

Comment:
Is there any way how I can convert this directly in a XMLRobertaTokenizerFast? I know I can use the old one here and then when loading the tokenizer from the hf checkpoint it can be converted on the fly.
But I think it would be preferable to convert it directly here.